acceptance/acceptance_test.go

```diff
@@ -1289,31 +1276,91 @@ func testAcceptance(
 								h.AssertEq(t, output, expectedOutput)
 							}
 						})
+
+						when("builder is untrusted", func() {
+							it("uses the 5 phases", func() {
+								var buf bytes.Buffer
+								var cmd *exec.Cmd
+								cmd = subjectPack(
+									"build", repoName,
+									"--builder", builderName, // untrusted by default
+									"-p", filepath.Join("testdata", "mock_app"),
+									"--publish",
+									"--network", "host",
+								)
+
+								cmd.Stdout = &buf
+								cmd.Stderr = &buf
+
+								h.AssertNil(t, cmd.Start())
+
+								go terminateAtStep(t, cmd, &buf, "[detector]")
+								err := cmd.Wait()
+								h.AssertNotNil(t, err)
+
+								h.AssertContains(t, buf.String(), "[detector]")
+
+								if !lifecycleImageSupported {
+									h.AssertContains(t, buf.String(), "Lifecycle does not have an associated lifecycle image")
+								}
+							})
+						})
+
+						when("builder is trusted", func() {
+							it("uses the creator (when supported)", func() {
+								h.SkipIf(t, !packSupports(packPath, "build --trust-builder"), "trust-builder not supported")
+
+								var buf bytes.Buffer
+								var cmd *exec.Cmd
+								cmd = subjectPack(
+									"build", repoName,
+									"--builder", builderName, // untrusted by default
+									"-p", filepath.Join("testdata", "mock_app"),
+									"--publish",
+									"--network", "host",
+									"--trust-builder",
+								)
+
+								cmd.Stdout = &buf
+								cmd.Stderr = &buf
+
+								h.AssertNil(t, cmd.Start())
+
+								if creatorSupported {
+									go terminateAtStep(t, cmd, &buf, "[creator]")
+									err := cmd.Wait()
+									h.AssertNotNil(t, err)
+
+									h.AssertContains(t, buf.String(), "[creator]")
+								} else {
+									go terminateAtStep(t, cmd, &buf, "[detector]")
+									err := cmd.Wait()
+									h.AssertNotNil(t, err)
+
+									h.AssertContains(t, buf.String(), "[detector]")
+								}
+							})
+						})
 					})
 
 					when("ctrl+c", func() {
 						it("stops the execution", func() {
 							var buf bytes.Buffer
-							cmd := subjectPack("build", repoName, "-p", filepath.Join("testdata", "mock_app"))
+							var cmd *exec.Cmd
```

Comment:
These weird declarations are an outgrowth of previous work that was reversed - I can put it back the way it was.